fed/proxy/barriers.py

```diff
@@ -223,43 +247,175 @@ def _start_sender_proxy(
+        self._party = party
+        self._tls_config = tls_config
+        job_config = fed_config.get_job_config()
+        cross_silo_comm_config = job_config.cross_silo_comm_config_dict
+        self._proxy_instance = proxy_cls(
+            addresses, party, tls_config, cross_silo_comm_config
+        )
+
+    def is_ready(self):
+        return self._proxy_instance.is_ready()
+
+    def start(self):
+        self._proxy_instance.start()
+
+    def get_data(self, src_party, upstream_seq_id, curr_seq_id):
+        self._stats["receive_op_count"] += 1
+        data = self._proxy_instance.get_data(src_party, upstream_seq_id, curr_seq_id)
+        return data
+
+    def send(
+        self,
+        dest_party,
+        data,
+        upstream_seq_id,
+        downstream_seq_id,
+    ):
+        self._stats["send_op_count"] += 1
+        assert (
+            dest_party in self._addresses
+        ), f'Failed to find {dest_party} in cluster {self._addresses}.'
+        send_log_msg = (
+            f'send data to seq_id {downstream_seq_id} of {dest_party} '
+            f'from {upstream_seq_id}'
+        )
+        logger.debug(
+            f'Sending {send_log_msg} with{"out" if not self._tls_config else ""}'
+            ' credentials.'
+        )
+        try:
+            response = self._proxy_instance.send(
+                dest_party, data, upstream_seq_id, downstream_seq_id
+            )
+        except Exception as e:
+            logger.error(f'Failed to {send_log_msg}, error: {e}')
+            return False
+        logger.debug(f"Succeeded to send {send_log_msg}. Response is {response}")
+        return True  # True indicates it's sent successfully.
+
+    def _get_stats(self):
+        return self._stats
+
+    def _get_proxy_config(self, dest_party=None):
+        return self._proxy_instance.get_proxy_config(dest_party)
+
+
+def _start_sender_receiver_proxy(
+    addresses: str,
+    party: str,
+    logging_level: str,
+    tls_config=None,
+    proxy_cls=None,
+    proxy_config: Dict = None,
+    ready_timeout_second: int = 60,
+):
+    global _DEFAULT_SENDER_RECEIVER_PROXY_OPTIONS
+    actor_options = copy.deepcopy(_DEFAULT_SENDER_RECEIVER_PROXY_OPTIONS)
+    if proxy_config:
+        proxy_config = fed_config.CrossSiloMessageConfig.from_dict(proxy_config)
+        if proxy_config.proxy_max_restarts:
+            actor_options.update(
+                {
+                    "max_task_retries": proxy_config.proxy_max_restarts,
+                    "max_restarts": 1,
+                }
+            )
+        if proxy_config.max_concurrency:
+            actor_options.update({"max_concurrency": proxy_config.max_concurrency})
+
+    logger.debug(f"Starting ReceiverProxyActor with options: {actor_options}")
+
+    global _SENDER_RECEIVER_PROXY_ACTOR
+    global _RECEIVER_PROXY_ACTOR_NAME
+    _SENDER_RECEIVER_PROXY_ACTOR = SenderReceiverProxyActor.options(
+        name=_RECEIVER_PROXY_ACTOR_NAME, **actor_options
```

Comment:
If this is named with receiver, how `send` get the SendProxyActor?

Reply:
When `SenderReceiverProxy` is enabled,  actor names of sender and receiver are same.